packages/js/components/src/media-uploader/stories/index.tsx

```diff
@@ -0,0 +1,164 @@
+/**
+ * External dependencies
+ */
+import React, { createElement } from 'react';
+import { Card, CardBody, Modal, Notice } from '@wordpress/components';
+import { MediaItem } from '@wordpress/media-utils';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { MediaUploader } from '../';
+import { ErrorType, File } from '../types';
+
+const MockMediaUpload = ( { onSelect, render } ) => {
+	const [ isOpen, setOpen ] = useState( false );
+
+	return (
+		<>
+			{ render( {
+				open: () => setOpen( true ),
+			} ) }
+			{ isOpen && (
+				<Modal
+					title="Media Modal"
+					onRequestClose={ () => setOpen( false ) }
+				>
+					<p>
+						Use the default built-in{ ' ' }
+						<code>MediaUploadComponent</code> prop to render the WP
+						Media Modal.
+					</p>
+					{ Array( ...Array( 3 ) ).map( ( n, i ) => {
+						return (
+							<button
+								key={ i }
+								onClick={ () => {
+									onSelect( {
+										alt: 'Random',
+										url: `https://picsum.photos/200?i=${ i }`,
+									} );
+									setOpen( false );
+								} }
+								style={ {
+									marginRight: '16px',
+								} }
+							>
+								<img
+									src={ `https://picsum.photos/200?i=${ i }` }
+									alt="Random"
+									style={ {
+										maxWidth: '100px',
+									} }
+								/>
+							</button>
+						);
+					} ) }
+				</Modal>
+			) }
+		</>
+	);
+};
+
+const ImageGallery = ( { images }: { images: File[] } ) => {
+	return (
+		<div style={ { marginBottom: '16px' } }>
+			{ images.map( ( image, index ) => {
+				return (
+					<img
+						key={ index }
+						alt={ image.alt }
+						src={ image.url }
+						style={ {
+							maxWidth: '100px',
+							marginRight: '16px',
+						} }
+					/>
+				);
+			} ) }
+		</div>
+	);
+};
+
+const mockUploadMedia = ( { filesList, onFileChange } ) =>
+	new Promise< void >( ( resolve ) => {
+		const fileReader = new FileReader();
+		fileReader.onload = function ( event ) {
+			const file = {
+				alt: 'Temporary image',
+				url: event?.target?.result,
+			} as MediaItem;
+			onFileChange( [ file ] );
+			resolve();
+		};
+		fileReader.readAsDataURL( filesList[ 0 ] );
+	} );
+
+export const Basic: React.FC = () => {
+	const [ images, setImages ] = useState< File[] >( [] );
+
+	return (
+		<Card size="large">
+			<CardBody>
+				<ImageGallery images={ images } />
+				<MediaUploader
+					MediaUploadComponent={ MockMediaUpload }
+					onSelect={ ( file ) => setImages( [ ...images, file ] ) }
+					onError={ () => null }
+					onUpload={ ( files ) =>
+						setImages( [ ...images, ...files ] )
```

Comment:
When I select multiple files and drag them in, I only get one file in the `files` array. This does appear to just be an issue with this story, because when I test in a WooCommerce admin page I get multiple firings of this callback, which eventually includes all three files I dragged.

Reply:
Ya, I was being lazy by not wanting to mock out a full uploader, but went ahead and fleshed it out for clarity.